storage/engine.go

```diff
@@ -129,6 +152,49 @@ func (e *engine) GetSessionDatabase() SessionDatabase {
 	return e.sessionDatabase
 }
 
+func (e *engine) SQLDatabase() *gorm.DB {
+	return e.sqlDB
+}
+
+// initSQLDatabase initializes the SQL database connection.
+// If the connection string is not configured, it defaults to a SQLite database, stored in the node's data directory.
+// Note: only SQLite is supported for now
+func (e *engine) initSQLDatabase() error {
+	connectionString := e.config.SQL.ConnectionString
+	if len(connectionString) == 0 {
+		connectionString = "file:" + path.Join(e.datadir, "sqlite.db")
+	}
+	var err error
+	e.sqlDB, err = gorm.Open(sqlite.Open(connectionString), &gorm.Config{})
+	if err != nil {
+		return err
+	}
+	log.Logger().Debug("Running database migrations...")
+	underlyingDB, err := e.sqlDB.DB()
+	if err != nil {
+		return err
+	}
+	sourceDriver, err := iofs.New(sqlMigrationsFS, "sql_migrations")
```

Comment:
How will the correct migrations be selected in the future when there's only a connection string? eg: mysql vs postgresql

Reply:
probably connection string driver name to folder